### ISSUE TITLE
Dev server must be restarted to have classes applied with Snowpack

### ISSUE THREAD

**Issue by samueldrai** [tailwindlabs/tailwindcss#3950]
### What version of @tailwindcss/jit are you using?

0.1.4

### What version of Node.js are you using?

14.2.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### Reproduction repository

https://www.youtube.com/watch?v=G6d-SXdcJXs

Hi there ☺️

I tried to use this new JIT feature with a new React app built with Snowpack. 
Unfortunately, I cannot have Snowpack use Tailwind JIT properly. 

I think I might have missed something about the configuration process.
When changing a class in a component, I have to stop and restart the dev server. 
Here are my config files: 

```javascript
// snowpack.config.js
module.exports = {
  mount: {
    public: { url: '/', static: true },
    src: { url: '/dist' },
  },
  plugins: [
    '@snowpack/plugin-react-refresh',
    '@snowpack/plugin-dotenv',
    '@snowpack/plugin-typescript',
    '@snowpack/plugin-postcss',
  ],
  routes: [{ match: 'routes', src: '.*', dest: '/index.html' }],
  optimize: {},
  packageOptions: {},
  devOptions: {},
  buildOptions: {},
}
```

```javascript
// tailwind.config.js
module.exports = {
  purge: ['./src/**/*.{js,jsx,ts,tsx}', './public/index.html'],
  darkMode: false,
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
}
```

```javascript
// postcss.config.js
module.exports = {
  plugins: {
    '@tailwindcss/jit': {},
    autoprefixer: {},
  },
}

```

Thanks!

**Comment by sebastiandittrich** [comment]
> I've published new Snowpack plugin that should help with this issue without any extra configuration. Can you guys test it if it works for you too? https://github.com/jadex/snowpack-plugin-tailwindcss-jit

This works for me, good work, thanks!

**Comment by mustafa0x** [comment]
I have this issue with rollup. `TAILWIND_MODE='watch'` didn't work. What does work is `touch tailwind.config.cjs` after each change. Maybe that indicates where the problem lies.

**Comment by mrodrigues95** [comment]
> 
> 
> I have this issue with rollup. `TAILWIND_MODE='watch'` didn't work. What does work is `touch tailwind.config.cjs` after each change. Maybe that indicates where the problem lies.

This is happening to me as well on a Next JS project running inside Docker. The only way for my changes to be updated is by opening up `tailwind.config.js` and hitting save.

**Comment by adamwathan** [comment]
@mustafa0x @mrodrigues95 can you both try setting TAILWIND_DISABLE_TOUCH=true? It will use an alternative watch method that is actually a bit better but doesn't work with all build tools (works with webpack and Rollup though).

**Comment by mustafa0x** [comment]
Thanks, that did it!

**Comment by mrodrigues95** [comment]
@adamwathan In my case, setting `CHOKIDAR_USEPOLLING=true` fixed it (along with `TAILWIND_MODE=watch`) since I am working inside a docker container.

**Comment by Xeevis** [comment]
@adamwathan I believe this issue can be closed, Snowpack has added official support for TailwindCSS in version `3.5.0+`
https://github.com/snowpackjs/snowpack/pull/3309
https://github.com/snowpackjs/snowpack/pull/3326

Install instructions available in the docs [https://www.snowpack.dev/guides/tailwind-css](https://www.snowpack.dev/guides/tailwind-css)

**Comment by teenjuna** [comment]
@Xeevis what's your configuration? For me it works only after manual page refresh and only when developer tools are open (Chrome).
I have this in package.json:
```
{
  "scripts": {
    "dev": "TAILWIND_MODE=watch NODE_ENV=development snowpack dev",
    "build": "TAILWIND_MODE=build NODE_ENV=production snowpack build",
  },
  "dependencies": {
    "@tailwindcss/forms": "^0.3.2",
    "solid-app-router": "^0.0.40",
    "solid-js": ">= 0.26"
  },
  "devDependencies": {
    "@babel/core": "^7.13.14",
    "@babel/preset-typescript": "^7.13.0",
    "@snowpack/plugin-babel": "^2.1.7",
    "@snowpack/plugin-dotenv": "^2.1.0",
    "@snowpack/plugin-postcss": "^1.4.0",
    "@snowpack/plugin-typescript": "^1.2.1",
    "@types/snowpack-env": "^2.3.3",
    "autoprefixer": "^10.2.5",
    "babel-preset-solid": "^0.25.0",
    "postcss": "^8.3.0",
    "postcss-cli": "^8.3.1",
    "prettier": "^2.2.1",
    "snowpack": "^3.5.1",
    "tailwindcss": "^2.1.2",
    "typescript": "^4.2.4"
  }
}
```
This is snowpack.config.js:
```
/**
 * @type {import('snowpack').SnowpackConfig}
 */
const config = {
  mount: {
    public: '/',
    src: '/assets',
  },
  packageOptions: {
    installTypes: true,
    NODE_ENV: true,
  },
  devOptions: {
    out: 'dist',
    open: 'none',
    bundle: true,
    tailwindConfig: './tailwind.config.js',
  },
  buildOptions: {
    clean: true,
    out: 'dist',
  },
  plugins: [
    '@snowpack/plugin-typescript',
    '@snowpack/plugin-babel',
    '@snowpack/plugin-postcss',
    '@snowpack/plugin-dotenv',
  ],
  optimize: {
    bundle: true,
    minify: true,
    target: 'es2020',
    treeshake: true,
    splitting: true,
  },
  alias: {
    '@': './src',
  },
  routes: [{ match: 'routes', src: '.*', dest: '/index.html' }],
};

module.exports = config;
```
This in tailwind.config.js:
```
const colors = require('tailwindcss/colors');

module.exports = {
  mode: 'jit',
  purge: ['src/**/*.tsx', 'public/index.html'],
  dark: false,
  theme: {
    extend: {
      colors: {
        gray: colors.trueGray,
      },
    },
  },
  variants: {},
  plugins: [require('@tailwindcss/forms')],
};
```
And this in postcss.config.js:
```
module.exports = {
  plugins: {
    tailwindcss: {},
  },
};
```

**Comment by Xeevis** [comment]
@teenjuna I had added `TAILWIND_DISABLE_TOUCH=true` (doesn't work without it for me)
```json
"scripts": {
  "dev": "TAILWIND_MODE=watch NODE_ENV=development TAILWIND_DISABLE_TOUCH=true snowpack dev",
},
```

I also had to upgrade to TailwindCSS 2.2.0 canary for it work for other markup files for purge other than `.html`

```bash
npm install -D tailwindcss@canary
```

Works like a charm
![image](https://user-images.githubusercontent.com/5835044/119326851-a54d2980-bc82-11eb-95ac-7a507c6b2159.png)

**Comment by teenjuna** [comment]
@Xeevis Interesting... This combination creates some kind of infinite self-triggering loop on my machine
![image](https://user-images.githubusercontent.com/53595243/119328545-6b921800-bcae-11eb-8d85-0d9305555b68.png)
...which is impossible to stop with Ctrl+C (kill -9 only)
Seems like an issue with canary release. And adding `TAILWIND_DISABLE_TOUCH` to the latest stable release doesn't help. So I'm not sure that this thread should be closed @adamwathan.

**Comment by bradlc** [comment]
Hey all. `@snowpack/plugin-postcss` was updated last week and here's how we recommend using Tailwind JIT with Snowpack:

1. Ensure that you are using the latest `tailwindcss` canary release (>= `2.2.0-canary.9`): `npm install -D tailwindcss@canary`
2. Ensure that you are using the latest version of `@snowpack/plugin-postcss` (>= `1.4.0`)
3. Set the `TAILWIND_DISABLE_TOUCH` environment variable to `true`. This enables an alternative watch method which is now compatible with `@snowpack/plugin-postcss`
4. Ensure that you are **_not_** using the `devOptions.tailwindConfig` Snowpack option

Here is a bare-bones example project: https://github.com/bradlc/jit-snowpack

If you are still having problems after following these instructions please open a new issue with a reproduction, thanks!

**Comment by adamburgess** [comment]
Repro of @teenjuna's issue: https://github.com/adamburgess/jit-snowpack
This is because tailwind tells snowpack to watch the entire dir.
So if your styles.css is in the directory that tailwind is set to purge on (note: tailwind does not pass the glob, only the directory... so even if you don't purge *.css your css **will** be included in this list), when you update styles.css, snowpack will see that styles.css has updated, go through all the dependencies (which _includes_ styles.css) and updates them... leading to infinite loop.

Fix is to either
1. Make tailwind send the actual glob through so the css doesn't depend on itself, or
2. Make @snowpack/plugin-postcss not mark itself as changed

**Comment by loctran016** [comment]
Any helps? Even though I've set like @bradlc said, it doesn't work, no errors in the console, but I still can't use tailwindcss. I've run `cross-env NODE_ENV=development TAILWIND_DISABLE_TOUCH=true yarn clean && npx @11ty/eleventy && snowpack dev`:

![image](https://user-images.githubusercontent.com/64778285/123102667-80a1c880-d45f-11eb-826b-3e85d1fb135a.png)

My repo is [here](https://github.com/loctran016/eleventy-kickstart-blog/tree/snowpack) (snowpack branch)

**Comment by adamwathan** [comment]
Hey! Please open a new issue like @bradlc mentioned in his last comment, it's the only way we won't lose track of it as we can't easily monitor closed issues.

**Comment by loctran016** [comment]
> Hey! Please open a new issue like @bradlc mentioned in his last comment, it's the only way we won't lose track of it as we can't easily monitor closed issues.

Oh, ok
